lib/mix/lib/mix.ex

```diff
@@ -705,7 +716,7 @@ defmodule Mix do
       end)
 
     config = Keyword.get(opts, :config, [])
-    config_path = opts[:config_path] && Path.expand(opts[:config_path])
+    config_path = expand_path(opts[:config_path], deps, :config_path)
```

Comment:
Tiny nitpick: I would pass the inner path here, so the function doesn't have to know the keys given to the caller:

```suggestion
    config_path = expand_path(opts[:config_path], deps, :config_path, "config/config.exs")
```

Reply:
This was my original implementation, but we'll actually need it for the error message

Reply:
I mean, pass both the key and the default value. :)